src/Pmi/Service/PatientStatusService.php

```diff
@@ -22,35 +25,75 @@ public function sendPatientStatusToRdr()
-                $this->app->log(Log::PATIENT_STATUS_HISTORY_EDIT, [
-                    'id' => $patientStatusHistory['id']
-                ]);
+                if ($patientStatusObj->saveData()) {
+                    $this->em->getRepository('patient_status_import_rows')->update($patientStatus['id'], ['rdr_status' => PatientStatusHistory::STATUS_SUCCESS]);
+                }
             } else {
-                $this->app['logger']->error("#{$patientStatusHistory['id']} failed sending to RDR: " . $this->rdr->getLastError());
+                $this->app['logger']->error("#{$patientStatus['id']} failed sending to RDR: " . $this->rdr->getLastError());
+                $rdrStatus = PatientStatusHistory::STATUS_OTHER_RDR_ERRORS;
+                if ($this->rdr->getLastErrorCode() === 400) {
+                    $rdrStatus = PatientStatusHistory::STATUS_INVALID_PARTICIPANT_ID;
+                } elseif ($this->rdr->getLastErrorCode() === 500) {
+                    $rdrStatus = PatientStatusHistory::STATUS_RDR_INTERNAL_SERVER_ERROR;
+                }
+                $this->em->getRepository('patient_status_import_rows')->update($patientStatus['id'], ['rdr_status' => $rdrStatus]);
+            }
+        }
+        // Update import status
+        $this->updateImportStatus($importIds);
+    }
+
+    private function updateImportStatus($importIds)
+    {
+        foreach ($importIds as $importId) {
+            $patientStatusImportRows = $this->em->getRepository('patient_status_import_rows')->fetchBySql('import_id = :importId and rdr_status = :rdrStatus', [
+                'importId' => $importId,
+                'rdrStatus' => 0
+            ]);
+            if (count($patientStatusImportRows) === 0) {
+                $patientStatusImportRows = $this->em->getRepository('patient_status_import_rows')->fetchBySql('import_id = :importId and rdr_status = (:rdrStatus)', [
+                    'importId' => $importId,
+                    'rdrStatus' => '2, 3, 4'
+                ]);
+                if (count($patientStatusImportRows) > 0) {
+                    $this->em->getRepository('patient_status_import')->update($importId, ['import_status' => PatientStatusImport::COMPLETE_WITH_ERRORS]);
+                } else {
+                    $this->em->getRepository('patient_status_import')->update($importId, ['import_status' => PatientStatusImport::COMPLETE]);
+                }
+                $this->app->log(Log::PATIENT_STATUS_IMPORT_EDIT, $importId);
             }
         }
     }
+
+    public function deleteUnconfirmedImportData()
+    {
+        $date = (new \DateTime('UTC'))->modify('-1 hours');
+        $date = $date->format('Y-m-d H:i:s');
+        $this->db->query("DELETE psir FROM patient_status_import_rows psir inner join patient_status_import psi on psir.import_id = psi.id where psi.created_ts < '{$date}' and psi.confirm = 0");
```

Comment:
Should we delete the patient_status_import row, too?

Reply:
I thought we could track the exports that are not confirmed but I am fine with deleting.

Reply:
If that's intentional, that's fine with me. I just wanted to make sure it would never show up for the user after the import rows have been deleted.